stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -442,7 +449,18 @@ public void onMessage(RespT message) {
     @Override
     public void onClose(Status status, Metadata trailers) {
       if (status.isOk()) {
-        observer.onCompleted();
+        if (!streamingResponse) {
+          if (unaryMessage != null) {
+            observer.onNext(unaryMessage);
```

Comment:
I only just noticed, because it is a bit subtle, but we now need to handle exceptions from `onNext()`.

Previously if onNext() threw an exception, onMessage() would propagate the exception which would cancel the call and so onClose() would be called, which would call `onError()`.

And the code will need a bit of an ugly dance. Maybe something like:
```java
boolean noException = false;
try {
  observer.onNext(unaryMessage);
  noException = true;
} catch (Throwable t) {
  observer.onError(t);
}
if (noException) {
  observer.onCompleted();
}
```